### ISSUE TITLE
[bug] Disconnect wallet is broken in v4

### ISSUE THREAD

**Issue by ethan-auroria** [WalletConnect/web3modal#1878]
### Link to minimal reproducible example

The most basic JS setup will do

### Summary

Upon clicking Disconnect button in the modal, nothing really happens, the client is not disconnected.
If I use `watchAccount`, the callback will be invoked with the active account, `isConnected` remains true.
If I attempt to click on the Disconnect button again, it will say "Failed to disconnect".
No error logs output in the console.
The connection persists accross page refreshes.
The only way to disconnect is to do so in the wallet.

### List of related npm package versions

"@wagmi/connectors": "^4.1.13",
"@wagmi/core": "^2.6.4",
"@web3modal/wagmi": "^4.0.0",
"viem": "^2.7.6",

**Comment by glitch-txs** [comment]
Does this happen when connecting with an extension wallet?

Are you able to repro here?
https://lab.web3modal.com

**Comment by ethan-auroria** [comment]
I realized the situation is a little more complicated, and cannot be replicated using lab. So here is the reproductiong repo: https://github.com/ethan-auroria/w3mv4-reprod

What I do here is initialize and store the instance of wagmi config, web3modal and wallet client in a global store (a Svelte construct). It is supposed to be running on client side only, as SSR has been disabled for the page. Also, I have only tested it with MetaMask browser extension, on Firefox and Chrome.
